modules/mqtt/destination/mqtt-grammar.ym

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2021 One Identity
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+
+%code requires {
+
+#include "mqtt-parser.h"
+
+}
+
+%code {
+
+#include "mqtt-destination.h"
+
+#include "cfg-grammar-internal.h"
+#include "plugin.h"
+}
+
+%define api.prefix {mqtt_destination_}
+%lex-param {CfgLexer *lexer}
+%parse-param {CfgLexer *lexer}
+%parse-param {LogDriver **instance}
+%parse-param {gpointer arg}
+
+/* INCLUDE_DECLS */
+
+%token KW_MQTT
+%token KW_HOST
+%token KW_PORT
+%token KW_TOPIC
+%token KW_CLEAN_SESSION
+%token KW_KEEPALIVE
+%token KW_QOS
+%token KW_PROTOCOL_TYPE
+%token KW_MESSAGE_TEMPLATE
+
+%%
+
+start
+        : LL_CONTEXT_DESTINATION KW_MQTT
+          {
+            last_driver = *instance = mqtt_destination_dd_new(configuration);
+          }
+          '(' mqtt_destination_options ')' { YYACCEPT; }
+;
+
+mqtt_destination_options
+        : mqtt_destination_option mqtt_destination_options
+        |
+        ;
+
+mqtt_destination_option
+        : KW_HOST '(' string ')'                { mqtt_destination_dd_set_host(last_driver, $3); free($3);  }
+        | KW_PORT '(' positive_integer ')'      { mqtt_destination_dd_set_port(last_driver, $3);            }
+        | KW_TOPIC '(' string ')'               { mqtt_destination_dd_set_topic(last_driver, $3); free($3); }
```

Comment:
topic should support templates as well. in a lot of cases mqtt is using device specific topics which would - in most cases - be dynamic and not a literal string.

Reply:
solved by this commit: MQTT/destination: topic can be templated  
(commit hash: 95e56c3888548e488a19c6c11780cfff4ec94371)